app/common/util/src/main/java/io/syndesis/common/util/Labels.java

```diff
@@ -53,30 +52,41 @@ public static String sanitize(String name) {
                         b.appendCodePoint(chr);
                     }
              }, StringBuilder::append)
-            .toString();
+            .toString());
     }
 
-    private static String trim(String name) {
-        String trimmed = name.length() > MAXIMUM_NAME_LENGTH ? name.substring(0, MAXIMUM_NAME_LENGTH) : name;
 
-        if (trimmed.matches(VALID_REGEX)) {
-            return trimmed;
+    public static boolean isValid(String name) {
+        return name.matches(VALID_REGEX) && name.length() <= MAXIMUM_NAME_LENGTH;
+    }
+
+    public static String validate(String name) {
+        if (!isValid(name)) {
+            throw new IllegalArgumentException("Invalid label: [" + name + "].");
+        }
+        return name;
+
+    }
+
+    private static String trim(String name) {
+        if (isValid(name)) {
+            return name;
         }
 
-        int length = trimmed.length();
+        int length = name.length();
         if (length <= 1) {
             throw new IllegalStateException("Specified string:" + name + " cannot be sanitized.");
         }
 
-        String first = trimmed.substring(0, 1);
+        String first = name.substring(0, 1);
         if (!first.matches(SIDES_REGEX)) {
-            return trim(trimmed.substring(1));
+            return trim(name.substring(1));
```

Comment:
Clever recursive call, but a bit expensive considering on regexp matching for a single char (maybe the trick as above just checking whether the charactes is unequal of .,- or _ (all other chars have been already filtered out before) is simpler ?  (and maybe even a while loop instead of going into recursion ?)